adi/cn0566.py

```diff
@@ -0,0 +1,620 @@
+
+                # Enable the Rx path for each channel
+                for channel in device.channels:
+                    channel.rx_enable = True  # this writes reg0x2E with data 0x00, then reg0x2E with data 0x20.
+                    channel.rx_gain = 127
+                    #  So it overwrites 0x2E, and enables only one channel
+
+            # Configure the device for Tx mode
+            elif self.device_mode == "tx":
+                device.mode = "tx"
+
+                # Enable the Tx path for each channel and set the external PA bias
+                for channel in device.channels:
+                    channel.tx_enable = True
+                    channel.tx_gain = 127
+                    channel.pa_bias_on = -2
+
+            else:
+                raise ValueError(
+                    "Configure Device in proper mode"
+                )  # If device mode is neither Rx nor Tx
+
+            if self.device_mode == "rx":
+                device.latch_rx_settings()  # writes 0x01 to reg 0x28.
+            elif self.device_mode == "tx":
+                device.latch_tx_settings()  # writes 0x02 to reg 0x28.
+
+    def save_channel_cal(self, filename="channel_cal_val.pkl"):
+        """ Saves channel calibration file."""
+        with open(filename, "wb") as file1:
+            pickle.dump(self.ccal, file1)  # save calibrated gain value to a file
+            file1.close()
+
+    def save_gain_cal(self, filename="gain_cal_val.pkl"):
+        """ Saves gain calibration file."""
+        with open(filename, "wb") as file1:
+            pickle.dump(self.gcal, file1)  # save calibrated gain value to a file
+            file1.close()
+
+    def save_phase_cal(self, filename="phase_cal_val.pkl"):
+        """ Saves phase calibration file."""
+        with open(filename, "wb") as file:
+            pickle.dump(self.pcal, file)  # save calibrated phase value to a file
+            file.close()
+
+    def load_channel_cal(self, filename="channel_cal_val.pkl"):
+        """
+        Load channel gain compensation values, if not calibrated set all to 0.
+
+        Parameters
+        ----------
+        filename: string
+            Path/name of channel calibration file
+        """
+        try:
+            with open(filename, "rb") as file:
+                self.ccal = pickle.load(file)  # Load gain cal values
+        except Exception:
+            print("file not found, loading default (no channel gain compensation)")
+            self.ccal = [0.0] * 2
+
+    def load_gain_cal(self, filename="gain_cal_val.pkl"):
+        """Load gain calibrated value, if not calibrated set all channel gain to maximum.
+
+        Parameters
+        ----------
+        filename: type=string
+            Provide path of gain calibration file
+        """
+        try:
+            with open(filename, "rb") as file1:
+                self.gcal = pickle.load(file1)  # Load gain cal values
+        except Exception:
```

Comment:
Need explicit exception and not generic one

Reply:
Done.